clay/random.clj

```diff
@@ -1,11 +1,2254 @@
+^:kindly/hide-code
 (ns random
-  (:require [fastmath.random :as r]
-
+  {:clj-kondo/config '{:config-in-call {fastmath.dev.clay/symbol-info-table {:ignore [:unresolved-symbol]}}}}
+  (:require [fastmath.core :as m]
+            [fastmath.random :as r]
+            [fastmath.kernel :as k]
+            [scicloj.kindly.v4.kind :as kind]
+            [fastmath.dev.ggplot :as gg]
+            [fastmath.dev.clay :as utls]
             [fastmath.dev.codox :as codox]))
 
-
 ;; # Random {.unnumbered}
 
+;; Collection of functions which deal with randomness. There are four groups:
+
+;; * random number generation
+;; * probability distributions
+;; * sequences generators
+;; * noise (value, gradient, simplex)
+
+^:kindly/hide-code
+(kind/code "(require '[fastmath.random :as r])")
+
+;; Set initial seed
+(r/set-seed! 31337)
+
+;; ## Common functions
+
+;; List of the functions which work with PRNG and distribution objects:
+
+;; ::: {.callout-tip title="Defined functions"}
+;; * `irandom`, `lrandom`, `frandom`, `drandom`
+;; * `->seq`
+;; * `set-seed!`, `set-seed`
+;; :::
+
+;; ### Random number generation
+
+;; * First argument should be PRNG or distribution object
+;; * With no additional arguments, functions return:
+;;     * full range for integers and longs
+;;     * $[0,1)$ for floats and doubles
+;; * Only for PRNGs
+;;     * With one additional argument, number from $[0,max)$ range is returned
+;;     * With two additional arguments, number form $[min,max)$ range is returned
+
+(utls/symbol-info-table
+ [[irandom "random integer, returns long"]
+  [lrandom "random long"]
+  [frandom "random float, returns boxed float"]
+  [drandom "random double"]])
+
+;; ### Sampling
+
+;; To generate infinite lazy sequence of random values, call `->seq` on given PRNG or distribution.
+;; Second (optional) argument can limit number of returned numbers.
+;; Third (optional) argument controls sampling method (for given `n` number of samples), there are the following options:
+
+;; * `:antithetic` - random values in pairs `[r1,1-r1,r2,1-r2,...]`
+;; * `:uniform` - spacings between numbers follow uniform distribution
+;; * `:systematic` - the same spacing with random starting point
+;; * `:stratified` - divide $[0,1]$ into `n` intervals and get random value from each subinterval
+
+(utls/symbol-info-table
```

Comment:
Same here, we don't need it (the same comment for the rest `symbol-info-table`)

Reply:
it's been removed.